hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/BaseFileOnlyRelation.scala

```diff
@@ -149,27 +152,10 @@ case class BaseFileOnlyRelation(override val sqlContext: SQLContext,
     val enableFileIndex = HoodieSparkConfUtils.getConfigValue(optParams, sparkSession.sessionState.conf,
       ENABLE_HOODIE_FILE_INDEX.key, ENABLE_HOODIE_FILE_INDEX.defaultValue.toString).toBoolean
     if (enableFileIndex && globPaths.isEmpty) {
-      // NOTE: There are currently 2 ways partition values could be fetched:
-      //          - Source columns (producing the values used for physical partitioning) will be read
-      //          from the data file
-      //          - Values parsed from the actual partition path would be appended to the final dataset
-      //
-      //        In the former case, we don't need to provide the partition-schema to the relation,
-      //        therefore we simply stub it w/ empty schema and use full table-schema as the one being
-      //        read from the data file.
```

Comment:
Is the first case a query optimization before?

Reply:
When converting `BaseFileRelation` to `HadoopFsRelation`, we must specifying partitionSchema to make some optimizer rules in Spark take effect such as `PruneFileSourcePartitions`. 

And in current code it will always provide partitionSchema if turns to `HadoopFsRelation`, so we simplify this code because `shouldExtractPartitionValuesFromPartitionPath` is always true in this case.

Reply:
Can you ensure that HUDI-4161 been soved after this change? Can you elaborate why `shouldExtractPartitionValuesFromPartitionPath` should be false after schema evolution?

Reply:
@wecharyu `shouldExtractPartitionValuesFromPartitionPath` can still return `false` based on super class?

Reply:
@danny0405 @yihua This PR followed HUDI-4161, the only difference is that when `hasSchemaOnRead` is true it migrates the `shouldExtractPartitionValuesFromPartitionPath` value from super class while the value is false before this pr. So the optimization in HUDI-4161 still works.

Also we do not aim to make `shouldExtractPartitionValuesFromPartitionPath` alway be false after schema evolution, after all hudi supports read partition values from both path and files. It should be determined by configurations and get from super class.

Reply:
But after your change, the partiton shema is always resolved from partiton path which looks like a regression?

Reply:
Got your point. The change here is because baseRelation will be converted to HadoopFsRelation only when `baseRelation.hasSchemaOnRead` is **false**:
https://github.com/apache/hudi/blob/65dd645b487a61fbca7e4e4b849d1f2f1ec143f9/hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DefaultSource.scala#L328-L332

In this case `shouldExtractPartitionValuesFromPartitionPath` is true, this is just a code simplify.

Reply:
Can you add this details info as a comment there.